### PR TITLE
Support in-built Project.el

### PR DESCRIPTION
**Pull request by ChetanKoneru**
@4DA please review.

Introduce `eshell-toggle-find-project-root-package'.

Make `eshell-toggle-use-project-root' obsolete.

Make sure this change does not break users current config, so use `defalias'
Fix warning.
update README.

### CODE REVIEW

**Comment by ChetanKoneru** [comment]
@4DA friendly ping.

File: eshell-toggle.el [hunk 1]
```diff
@@ -127,21 +134,32 @@
...
 
 (defun eshell-toggle--make-buffer-name ()
   "Generate toggle buffer name."
-  (if eshell-toggle-use-projectile-root
-      (concat "*et" eshell-toggle-name-separator (projectile-project-name) "*")
-    (let* ((dir (eshell-toggle--get-directory))
-           (name (string-join (split-string dir "/") eshell-toggle-name-separator))
-           (buf-name (concat "*et" name "*")))
-      buf-name)))
+  (or
```
Comment (a13) [inline eshell-toggle.el]: better use `cond`, since the `else` parts of both `if` branches look the same

File: eshell-toggle.el [hunk 2]
```diff
@@ -74,6 +74,13 @@
                  (const :tag "Enabled" t))
   :group 'eshell-toggle)
 
+(defcustom eshell-toggle-use-project-root
```
Comment (a13) [inline eshell-toggle.el]: probably we should just add another option to `eshell-toggle-use-projectile-root`, smth like `nil` for "Disabled", `t` and `'projectile` for projectile, and `'project` to use project.el.

**Review by 4DA — CHANGES_REQUESTED** [review]
please follow @a13 advice: make add eshell-toggle-use-projectile-root accept different values: nil, t(for projectile), ‘projectile, ‘project.

also please refactor to cond clause on line 150 as advised

**Comment by brsvh** [comment]
@CsBigDataHub are you still continuing this work? I followed the review and created a revision, maybe you can pick my work, [patch here].

[patch here]: https://github.com/brsvh/eshell-toggle/compare/100e7069b428b2cf65b1d3cd987234e0b9d4deb3..6611888be2fe834f044db26436ac3f977858f5c8

**Comment by ChetanKoneru** [comment]
Updated.

Thank you @brsvh, @a13 , @4DA .

**Comment by ChetanKoneru** [comment]
> Updated.
> 
> Thank you @brsvh, @a13 , @4DA .

@4DA  friendly ping. Thank you.

**Comment by 4DA** [comment]
@CsBigDataHub merged, thank you for contribution!

**Pull request merged**